### PR TITLE
Plugin validation fixes

### PR DESCRIPTION
**Pull request by vsverchinsky**
Resolves: 

1. Startup plugin validation freeze
2. "New" radio button was available if `wxUSE_ACCESSIBILITY` was not defined

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: libraries/lib-module-manager/AsyncPluginValidator.cpp [hunk 1]
```diff
@@ -57,9 +57,8 @@ class AsyncPluginValidator::Impl final :
    {
       auto server = std::make_unique<IPCServer>(*this);
       if(!PluginHost::Start())
-         mDelegate->OnInternalError("Cannot start plugin host");
-      else
-         mServer = std::move(server);
+         throw std::runtime_error("cannot start plugin host process");
```
Comment (Paul-Licameli) [inline libraries/lib-module-manager/AsyncPluginValidator.cpp:60]: This possibly thrown exception is not caught anywhere and might terminate the program.  One remedy would be, to move initialization of PluginStartupRegistration::mValidator out of the constructor, and into the try-catch block in Run().

File: src/PluginStartupRegistration.cpp [hunk 2]
```diff
@@ -52,23 +53,32 @@ const std::vector<wxString>& PluginStartupRegistration::GetFailedPlugins() const
...
+         //delivery, but ProgressDialog::Poll implementation does not call
+         //wxApp::Yield each time, which may result in too long CPU stalls
+         if(wxTheApp->HasPendingEvents())
+            wxTheApp->Yield();
+         else
+            //Seems like we do have no events yet, save CPU cycles then
+            std::this_thread::sleep_for(std::chrono::milliseconds { 10 });
+      }
+   }
+   catch(std::exception& e)
```
Comment (Paul-Licameli) [inline src/PluginStartupRegistration.cpp:76]: Was this catch meant for the new throw?  But it does not enclose it.

**Review by Paul-Licameli — CHANGES_REQUESTED** [review]
try/catch is misplaced for the new exception

File: libraries/lib-module-manager/AsyncPluginValidator.cpp (hunk 1 above)
Comment (vsverchinsky) [reply libraries/lib-module-manager/AsyncPluginValidator.cpp:60]: StartHost isn't called from the constructor

File: src/PluginStartupRegistration.cpp (hunk 2 above)
Comment (vsverchinsky) [reply src/PluginStartupRegistration.cpp:76]: I don't see any other places from where StartHost could be called

**Comment by Paul-Licameli** [comment]
I was mistaken to trace the call graph of the constructor instead of StartHost.  However, is this an example of a call stack that could leave a potential exception uncaught, in one of the worker threads of `BufferedIPCChannel`?

```
Impl::StartHost
Impl::Validate
AsyncPluginValidator::Validate
PluginStartupRegistration::ProcessNext
PluginStartupRegistration::OnValidationFinished
AsyncPluginValidator::Impl::HandleResult
AsyncPluginValidator::Impl::OnDataAvailable or OnDisconnect
thread function spawned in BufferedIPCChannel::StartConversation
```

So, I think one more try-catch is needed to be sure exceptions from `Validate` are stopped.  It could perhaps be in `OnValidationFinished`

**Comment by vsverchinsky** [comment]
> So, I think one more try-catch is needed to be sure exceptions from `Validate` are stopped. It could perhaps be in `OnValidationFinished`

Where? All calls to AsyncPluginValidator::Delegate are dispatched to the main thread

**Comment by Paul-Licameli** [comment]
> > So, I think one more try-catch is needed to be sure exceptions from `Validate` are stopped. It could perhaps be in `OnValidationFinished`
> 
> Where? All calls to AsyncPluginValidator::Delegate are dispatched to the main thread

Do you mean the function `AsyncPluginValidator::Validate` ?
I see it is called by `PluginStartupRegistration::ProcessNext`, and that is called in two places, but only one is inside a try block.  The other is in `PluginStartupRegistration::OnValidationFinished`

Following the call stack up from there -- I was mistaken, what calls that is a lambda that the other thread enqueues for the main thread to do, using CallAfter.  Still:  in the main thread, there is this other place needing a try-catch to be sure the exception can't terminate the program.

**Comment by vsverchinsky** [comment]
> > > So, I think one more try-catch is needed to be sure exceptions from `Validate` are stopped. It could perhaps be in `OnValidationFinished`
> > 
> > 
> > Where? All calls to AsyncPluginValidator::Delegate are dispatched to the main thread
> 
> Do you mean the function `AsyncPluginValidator::Validate` ? I see it is called by `PluginStartupRegistration::ProcessNext`, and that is called in two places, but only one is inside a try block. The other is in `PluginStartupRegistration::OnValidationFinished`
> 
> Following the call stack up from there -- I was mistaken, what calls that is a lambda that the other thread enqueues for the main thread to do, using CallAfter. Still: in the main thread, there is this other place needing a try-catch to be sure the exception can't terminate the program.

The main thread execution happens inside `while` loop

**Review by Paul-Licameli — APPROVED** [review]

**Comment by Paul-Licameli** [comment]
I see and approve the latest rewrite, which makes one narrow-scope try-catch block at a lower level, sufficient to catch all the exceptions.

Now I ask you to review the suggested extra commit from me and merge it, if it helps to avoid traps in the debug build.

You may also notice -- the added use of `Translation()` suggests that plain-UI dialogs may fail to update a localized string in case the user happens to change the language preference.  That is true; that is a common problem with plain VST and Ladspa dialogs; that could also be fixed as in commit 20ace31; but, it is an unlikely use case and is beside the point of the fix required just now.

**Pull request merged**